settings.cmake

```diff
@@ -175,4 +175,18 @@ if(NOT Sel4benchAllowSettingsOverride)
     endif()
 
     # Add new app-specific configuration here
+
+
+    # Benchmarks that don't have a timer can't use the benchmarks which need
+    # a timer.
+    if (LibPlatSupportNoPlatformLtimer)
```

Comment:
This option won't be defined until the utils_libs cmake module is imported later in the project. So at this point it may be better to make it a platform based test?

Reply:
CMake does process things procedurally but it has generator expressions that implement a generation phase processing that happens at the end of CMake processing. But the config system that the build uses doesn't use generator expressions and so all of the config options are evaluated in the order they're get and set.
But then because the config options are also stored in the cache (because they can be set by a user on the command line or interactively), running CMake a second time can sometimes hide configuration logic bugs because it uses the cache values from the last run.

Reply:
It's not very user friendly, I've wondered about dropping the caching from the options so that they're always evaluated fresh, but that then loses the ability to do the interactive config editing

Reply:
You could move this code to https://github.com/seL4/sel4bench/blob/master/apps/sel4bench/CMakeLists.txt#L63  so that its evaluated after the libraries are imported but before the apps are imported.

Reply:
The pattern of putting configs in settings.cmake used by sel4test and sel4bench is mainly for setting Kernel* config options and especially ones that impact toolchain selection because they have to be evaluated before the first call to `project()` that permanently selects the toolchains for the languages enabled.